### PR TITLE
add(delete entries before post processing)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

Adds functionality that was present in the demux post processing api but was not yet implemented.

### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/services/illumina_services/illumina_post_processing_service/illumina_post_processing_service.py [hunk 1]
```diff
@@ -183,6 +183,7 @@ def post_process_illumina_flow_cell(
         if self.dry_run:
             LOG.info(f"Dry run: will not post-process Illumina run {sequencing_run_name}")
             return
+        self.delete_sequencing_run_data(flow_cell_id=run_directory_data.id)
```
Comment (diitaz93) [inline cg/services/illumina_services/illumina_post_processing_service/illumina_post_processing_service.py:186]: as this raises an error, should this be inside the try-except?

File: tests/store/crud/delete/test_delete.py [hunk 2]
```diff
@@ -105,3 +105,40 @@ def test_store_api_delete_non_existing_case(
     remaining_cases: list[Case] = store_with_multiple_cases_and_samples.get_cases()
 
     assert len(remaining_cases) == len(existing_cases)
+
+
+def test_delete_illumina_flow_cell(
+    store_with_illumina_sequencing_data: Store, novaseq_x_flow_cell_id
```
Comment (diitaz93) [inline tests/store/crud/delete/test_delete.py:111]:
```suggestion
    store_with_illumina_sequencing_data: Store, novaseq_x_flow_cell_id: str
```

**Review by diitaz93 — APPROVED** [review]
Looks good, the only thing I would like to be addressed is to include the delete method inside the try

File: cg/services/illumina_services/illumina_post_processing_service/illumina_post_processing_service.py (hunk 1 above)
Comment (ChrOertlin) [reply cg/services/illumina_services/illumina_post_processing_service/illumina_post_processing_service.py:186]: The try and exept for the raised error is inside the function.

Comment (diitaz93) [reply cg/services/illumina_services/illumina_post_processing_service/illumina_post_processing_service.py:187]: ok, I will resolve this

**Pull request merged**